packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -805,6 +808,7 @@ class HeroController extends NavigatorObserver {
       if (toHeroes[tag] != null) {
         final HeroFlightShuttleBuilder fromShuttleBuilder = fromHeroes[tag].widget.flightShuttleBuilder;
         final HeroFlightShuttleBuilder toShuttleBuilder = toHeroes[tag].widget.flightShuttleBuilder;
+        final bool isDiverted = _flights[tag] != null;
```

Comment:
Line 832 has `_flights[tag] != null` as a condition as well, but since it's in a different if/else block, it means something completely different (`isAborted`). 

I'm not sure if this is a real issue or not, but the first time I looked at it, I was a little confused

Reply:
`_flights[tag] != null` means there's currently a hero in-flight with the given tag. If `toHeroes[tag] != null` we have a new `to` destination for that hero, so we divert it. If `toHeroes[tag] == null` the in-flight hero is no longer contained in the destination route. Therefore, we don't have a `to` destination for the flying hero anymore and we abort its flight. Hope this makes sense.